tests/tdastro/astro_utils/test_passbands.py

```diff
@@ -116,51 +102,164 @@ def mock_urlretrieve(url, filename, *args, **kwargs):  # Urlretrieve saves conte
 
     # Mock the urlretrieve portion of the download method
     with patch("urllib.request.urlretrieve", side_effect=mock_urlretrieve) as mocked_urlretrieve:
-        a_band = Passband(survey, band_label, table_path=table_path, table_url=table_url)
+        a_band = Passband(survey, filter_name, table_path=table_path, table_url=table_url)
 
         # Check that the transmission table was downloaded
         mocked_urlretrieve.assert_called_once_with(table_url, table_path)
 
         # Check that the transmission table was loaded correctly
-        assert a_band.transmission_table.shape == (3, 2)
-        assert a_band.transmission_table[0, 0] == 1000
-        assert a_band.transmission_table[0, 1] == 0.5
-        assert a_band.transmission_table[1, 0] == 1001
-        assert a_band.transmission_table[1, 1] == 0.6
-        assert a_band.transmission_table[2, 0] == 1002
-        assert a_band.transmission_table[2, 1] == 0.7
+        np.testing.assert_allclose(a_band._loaded_table, np.array([[1000, 0.5], [1005, 0.6], [1010, 0.7]]))
+
+
+def test_process_transmission_table(tmp_path):
+    """Test the process_transmission_table method of the Passband class; check correct methods are called."""
+    transmission_table = "100 0.5\n200 0.75\n300 0.25\n"
+    a_band = create_test_passband(tmp_path, transmission_table)
+
+    # Mock the methods that _process_transmission_table wraps, to check each is called once
+    with (
+        patch.object(a_band, "_interpolate_transmission_table") as mock_interp_table,
+        patch.object(a_band, "_trim_transmission_by_quantile") as mock_trim_table,
+        patch.object(a_band, "_normalize_transmission_table") as mock_norm_table,
+    ):
+        # Call the _process_transmission_table method
+        delta_wave, trim_quantile = 1.0, 0.05
+        a_band.process_transmission_table(delta_wave, trim_quantile)
+
+        # Check that each method is called once
+        mock_interp_table.assert_called_once_with(a_band._loaded_table, delta_wave)
+        mock_trim_table.assert_called_once()
+        mock_norm_table.assert_called_once()
+
+    # Now call without mocking, to check waves set correctly (other values checked in method-specific tests)
+    delta_wave, trim_quantile = 5.0, None
+    a_band.process_transmission_table(delta_wave, trim_quantile)
+    np.testing.assert_allclose(a_band.waves, np.arange(100, 301, delta_wave))
+
+
+def test_interpolate_transmission_table(tmp_path):
+    """Test the _interpolate_transmission_table method of the Passband class."""
+    transmission_table = "100 0.5\n200 0.75\n300 0.25\n"
+    a_band = create_test_passband(tmp_path, transmission_table)
```

Comment:
One advantage of allowing the constructor to take in a table (see suggestion in passbands.py) would be that you don't have to write out to files in each of these tests, which would speed things up.